controller/src/main/java/io/pravega/controller/server/rest/RESTServer.java

```diff
@@ -68,12 +77,43 @@ protected void startUp() {
         long traceId = LoggerHelpers.traceEnterWithContext(log, this.objectId, "startUp");
         try {
             log.info("Starting REST server listening on port: {}", this.restServerConfig.getPort());
-            httpServer = GrizzlyHttpServerFactory.createHttpServer(baseUri, resourceConfig, true);
+            if (restServerConfig.isTlsEnabled()) {
+                SSLContextConfigurator contextConfigurator = new SSLContextConfigurator();
+                contextConfigurator.setKeyStoreFile(restServerConfig.getKeyFilePath());
+                contextConfigurator.setKeyStorePass(loadPasswordFromFile(restServerConfig.getKeyFilePasswordPath()));
+                httpServer = GrizzlyHttpServerFactory.createHttpServer(baseUri, resourceConfig, true,
+                        new SSLEngineConfigurator(contextConfigurator, false, false, false));
+            } else {
+                httpServer = GrizzlyHttpServerFactory.createHttpServer(baseUri, resourceConfig, true);
+            }
         } finally {
             LoggerHelpers.traceLeave(log, this.objectId, "startUp", traceId);
         }
     }
 
+    private String loadPasswordFromFile(String keyFilePasswordPath) {
+        // In case the path is not specified, return empty string. This means the password is not used.
+        // In case of error (including very large password file), return empty password.
+        // Which will fail the SSL connection if the password is expected.
+
+        String password = "";
+        if (!Strings.isNullOrEmpty(keyFilePasswordPath)) {
+            File passwordFile = new File(keyFilePasswordPath);
+            if (passwordFile.length() != 0) {
+                if (passwordFile.length() > MAX_PASSWORD_LENGTH) {
+                    log.warn("Very large password file. Not parsing.");
+                } else {
+                    try {
+                        password = new String(FileUtils.readFileToByteArray(passwordFile), Charsets.UTF_8).trim();
```

Comment:
Why are we not propagating the exception? Shouldn't the application be made aware explicitly that there is a problem with the password file rather than proceed (almost) silently. The warn message is good, but this is important enough that the application needs to be warned explicitly.

Also, I assume that if we require a password, but we can't get it, then the server will fail because it won't be able to open the key file, yes? How's it going to fail in this case? In particular, what happens to this call:

https://github.com/pravega/pravega/pull/2587/files#diff-120984f086091c8ab4ed8a3cb8ed6cefR83

Reply:
Thinking about this again, I think the logic is that password is going to be empty and the SSL connection will fail because it expects one. I'd like to see a test case that covers this scenario, showing that the user can't bypass our security when using an invalid password file. Is large file the only way this can fail?

Reply:
Added a test case which validates that grizzly server does not start when the password does not match.